hyperspy/tests/test_events.py

```diff
@@ -0,0 +1,91 @@
+import nose.tools
+import hyperspy.events as he
+
+class TriggeredCorrectly: pass
+
+class TestEventsSuppression:
+        
+    def on_trigger(self, should_trigger):
+        if should_trigger:
+            raise TriggeredCorrectly()
+        else:
+            nose.tools.assert_true(False, "This event should not trigger")
+            
+    def trigger_check(self, trigger, should_trigger):
+        if should_trigger:
+            nose.tools.assert_raises(TriggeredCorrectly, trigger, True)
+        else:
+            trigger(False)
+        
+    def setUp(self):
+        self.events = he.Events()
+        
+    def reset(self):
```

Comment:
I you call this method `setUp` nose will call it for your every time that it runs a test so that you don't need to call it manually.